arduino/sketch/sketch.go

```diff
@@ -293,14 +291,15 @@ func CheckForPdeFiles(sketch *paths.Path) []*paths.Path {
 	return files
 }
 
-// GenBuildPath generates a suitable name for the build folder.
-// The sketchPath, if not nil, is also used to furhter differentiate build paths.
-func GenBuildPath(sketchPath *paths.Path) *paths.Path {
```

Comment:
wasn't a different hash created in the case a sketchPath was given? We don't need it anymore?

Reply:
No, actually the hash depends only on the given `sketchPath` (a.k.a. `sketch.FullPath`).

I removed the static function `GenBuildPath` and put it as a method of `Sketch` -> `sketch.DefaultBuildPath(...)` which is much more expressive.

Also, I've made the `sketch.Hash()` function that gives back only the "hash" piece so we can use it in #2033 as a key for a directory cache.